libcontainer/cgroups/systemd/v1.go

```diff
@@ -207,9 +207,10 @@ func (m *legacyManager) Destroy() error {
 
 	stopErr := stopUnit(m.dbus, getUnitName(m.cgroups))
 
-	// Both on success and on error, cleanup all the cgroups we are aware of.
-	// Some of them were created directly by Apply() and are not managed by systemd.
-	if err := cgroups.RemovePaths(m.paths); err != nil {
+	// Both on success and on error, cleanup all the cgroups
+	// we are aware of, as some of them were created directly
+	// by Apply() and are not managed by systemd.
+	if err := cgroups.RemovePaths(m.paths); err != nil && stopErr == nil {
```

Comment:
I feel this change could've been a bit neater but it works so w/e.

Reply:
You mean something like this?
```go
rmErr := cgroups.RemovePaths(m.paths)
// stopErr should prevail as it might be the reason for rmErr.
if stopErr != nil {
     return stopErr
}
return rmErr